### DISCUSSION TITLE
Organise Properties and Commands etc into Regions

### DISCUSSION THREAD

**Discussion by Montago** [codecadwallader/codemaid#846] [Ideas]
I would love to see some sort of option to organize intertwined code into regions, especially Properties, Commands  and maybe also Events

eg: 
```
        #region Property: IsUpdateLabelVisible 
        private Visibility _isUpdateLabelVisible = Visibility.Hidden;
        public Visibility IsUpdateLabelVisible
        {
            get { return _isUpdateLabelVisible; }
            set { _isUpdateLabelVisible = value; OnPropertyChanged(); }
        }
        #endregion
```

or 
```
        #region Command: MyAwesomeCommand
        private ICommand _awesome;

        public ICommand MyAwesomeCommand => _awesome = new RelayCommand(ExecuteMyAwesome, CanMyAwesome);

        public bool CanMyAwesome() => true;

        private async void ExecuteMyAwesome(object obj)
        {

        }
        #endregion
```

As you can see, Properties, Commands and sometimes Events that reference methods within the same Class belong together in a Region. 

It makes it easier to navigate the code - instead of lookin a big pile of [Public properties] or big pile of [Private fields]

Likewize the Regions should follow a naming convention eg: "Property : MyProperty" and "Command : MyCommand"

**Comment by Montago** [comment]
@codecadwallader  sorry for poking you directly .. but would this form of formatting be possible ? 

it would make Codemaid relevant for me again, since all the features i used is now built into Visual Studio :-/ 

It seems that the formatting options i either rather limited or require advanced knowledge how to setup properly.

**Reply by codecadwallader** [reply]
Thanks for the feature suggestion.  If you can create these under the Issues tab this is where these are usually located.  It is possible, but I do not know if anyone will pick it up for development.
